pa

```diff
@@ -56,16 +56,15 @@ pw_edit() {
     [ -w /dev/shm ] || tmpdir=/tmp
 
     # Reimplement mktemp here, because
-    # mktemp isn't defined in POSIX
-    editdir="$tmpdir/pa.$(rand_chars 8 '[:alnum:]')"
+    # mktemp isn't defined in POSIX.
+    suffix="$(rand_chars 10 'A-Za-z0-9')"
 
-    tmpfile="$editdir/$name.age"
+    tmpfile="$tmpdir/pa.$suffix/$name.age"
 
-    # Handle nested items (/foo/bar.age)
-    mkdir -p "$(dirname "$tmpfile")" ||
+    [ -z "$suffix" ] || ! mkdir -p "$(dirname "$tmpfile")" &&
```

Comment:
what is `[ -z "$suffix" ]` doing here? it looks a little out of place to me.

if it's testing whether `rand_chars` is returning a string, we should make that assertion in `rand_chars` instead, so that all calls are covered?

Reply:
if /dev/urandom is unavailable, `rand_chars` will return an empty string, so it's checking for that. but I recently found out that when assigning the output of a command to a variable like `var=$(cmd)`, the whole assignment will inherit status of the command, so I can do this instead:
```sh
suffix="$(rand_chars 10 'A-Za-z0-9')" ||
    die "Couldn't generate random characters"
```
it's better IMO than doing an assertion in `rand_chars`, I don't want to have a helper function like this call `die` directly. I think it's good that `die`s are reserved to `pw_*()` and `main()` functions now. in addition to this bit, `rand_chars` is only called by `pw_add()` and there's already a check like this.

Reply:
sick, that's perfect.